src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,44 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    static final int MIN_AGE = 18;
+    static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Login is null");
+        }
+
+        if (user.getLogin().isEmpty()) {
+            throw new RuntimeException("Login is empty");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("Login are not unique");
+        }
+        if (user.getAge() == null) {
+            throw new RuntimeException("Age is null");
+        }
+
+        if (user.getAge() < MIN_AGE) {
+            throw new RuntimeException("Age is lower than 18");
```

Comment:
Use constants in messages as well - constant can be changed during refactoring and message can become incorrect.